core/src/main/java/org/apache/accumulo/core/conf/SiteConfiguration.java

```
@@ -94,33 +123,45 @@ synchronized public static SiteConfiguration getInstance() {
-      if (configFile.startsWith("file://")) {
-        try {
-          File f = new File(new URI(configFile));
-          if (f.exists() && !f.isDirectory()) {
-            xmlConfig.addResource(f.toURI().toURL());
-            log.info("Loaded configuration from filesystem at {}", configFile);
-          } else {
-            log.warn("Failed to load Accumulo configuration from " + configFile, new Throwable());
-          }
-        } catch (MalformedURLException | URISyntaxException e) {
-          log.warn("Failed to load Accumulo configuration from " + configFile, e);
-        }
-      } else {
-        URL accumuloConfigUrl = SiteConfiguration.class.getClassLoader().getResource(configFile);
-        if (accumuloConfigUrl == null) {
-          log.warn("Accumulo configuration '" + configFile + "' is not on classpath",
-              new Throwable());
+  synchronized public static SiteConfiguration getInstance(File accumuloSiteFile) {
+    if (instance == null) {
+      instance = new SiteConfiguration(accumuloSiteFile);
+      ConfigSanityCheck.validate(instance);
+    }
+    return instance;
+  }
+
+  public static URL getAccumuloSiteLocation() {
+    String configFile = System.getProperty("accumulo.configuration", "accumulo-site.xml");
+    if (configFile.startsWith("file://")) {
+      try {
+        File f = new File(new URI(configFile));
+        if (f.exists() && !f.isDirectory()) {
+          log.info("Found Accumulo configuration at {}", configFile);
+          return f.toURI().toURL();
         } else {
-          xmlConfig.addResource(accumuloConfigUrl);
-          log.info("Loaded configuration from classpath at {}", accumuloConfigUrl.getFile());
+          log.warn("Failed to load Accumulo configuration at " + configFile, new Throwable());
         }
+      } catch (MalformedURLException | URISyntaxException e) {
+        log.warn("Failed to load Accumulo configuration from " + configFile, e);
```

This should be a hard fail if the configuration the user specified isn't available. We shouldn't fall back to magic behavior, since the configuration drives a lot of security-related elements.

---

fixed now